urllib3/poolmanager.py

```diff
@@ -40,38 +39,42 @@
 )
 
 
-def _default_key_normalizer(key_class, request_context):
+def _default_key_normalizer(request_context):
     """
-    Create a pool key of type ``key_class`` for a request.
+    Create a pool key out of a request context dictionary.
 
     According to RFC 3986, both the scheme and host are case-insensitive.
     Therefore, this function normalizes both before constructing the pool
     key for an HTTPS request. If you wish to change this behaviour, provide
     alternate callables to ``key_fn_by_scheme``.
 
-    :param key_class:
-        The class to use when constructing the key. This should be a namedtuple
-        with the ``scheme`` and ``host`` keys at a minimum.
-
     :param request_context:
         A dictionary-like object that contain the context for a request.
         It should contain a key for each field in the :class:`HTTPPoolKey`
     """
-    context = {}
-    for key in key_class._fields:
-        context[key] = request_context.get(key)
+    # Since we mutate the dictionary, make a copy first
+    context = request_context.copy()
     context['scheme'] = context['scheme'].lower()
     context['host'] = context['host'].lower()
-    return key_class(**context)
+
+    return _frozenset_from_dictionary(context)
```

Comment:
I mean, the restriction is implicit eventually because kwargs that are unrecognised at the pool level are passed to the connection, at which point they are either accepted or not.

However, I take your point. Hrm. I do still feel like the namedtuple is the right way to go, but the headers kwarg does rerepsent a problem.